src/pybind/mgr/dashboard/services/sso.py

```diff
@@ -174,43 +178,34 @@ def handle_sso_command(cmd):
         idp_metadata = cmd['idp_metadata']
         idp_username_attribute = _get_optional_attr(cmd, 'idp_username_attribute', 'uid')
         idp_entity_id = _get_optional_attr(cmd, 'idp_entity_id', None)
-        sp_x_509_cert = _get_optional_attr(cmd, 'sp_x_509_cert', '')
-        sp_private_key = _get_optional_attr(cmd, 'sp_private_key', '')
-        if sp_x_509_cert and not sp_private_key:
+        sp_x_509_cert_path = _get_optional_attr(cmd, 'sp_x_509_cert', '')
+        sp_private_key_path = _get_optional_attr(cmd, 'sp_private_key', '')
+        if sp_x_509_cert_path and not sp_private_key_path:
             return -errno.EINVAL, '', 'Missing parameter `sp_private_key`.'
-        if not sp_x_509_cert and sp_private_key:
+        if not sp_x_509_cert_path and sp_private_key_path:
             return -errno.EINVAL, '', 'Missing parameter `sp_x_509_cert`.'
-        has_sp_cert = sp_x_509_cert != "" and sp_private_key != ""
+        has_sp_cert = sp_x_509_cert_path != "" and sp_private_key_path != ""
         try:
-            f = open(sp_x_509_cert, 'r')
-            sp_x_509_cert = f.read()
-            f.close()
+            with open(sp_x_509_cert_path, 'r') as f:
+                sp_x_509_cert = f.read()
         except FileNotFoundError:
-            pass
+            sp_x_509_cert = ''
         try:
-            f = open(sp_private_key, 'r')
-            sp_private_key = f.read()
-            f.close()
+            with open(sp_private_key_path, 'r') as f:
+                sp_private_key = f.read()
         except FileNotFoundError:
-            pass
-        try:
-            idp_settings = OneLogin_Saml2_IdPMetadataParser.parse_remote(idp_metadata,
-                                                                         validate_cert=False,
-                                                                         entity_id=idp_entity_id)
-        # pylint: disable=broad-except
-        except Exception:
-            try:
-                f = open(idp_metadata, 'r')
-                idp_metadata = f.read()
-                f.close()
-            except FileNotFoundError:
-                pass
-            try:
-                idp_settings = OneLogin_Saml2_IdPMetadataParser.parse(idp_metadata,
-                                                                      entity_id=idp_entity_id)
-            # pylint: disable=broad-except
-            except Exception:
-                return -errno.EINVAL, '', 'Invalid parameter `idp_metadata`.'
+            sp_private_key = ''
+
+        if os.path.isfile(idp_metadata):
+            warnings.warn_explicit(
+                "Please prepend 'file://' to indicate a local SAML2 IdP file", DeprecationWarning)
+            with open(idp_metadata, 'r') as f:
```

Comment:
Why not appending ``file://``here and let fall through to the following ``parse.urlparse``code path?

Reply:
That could work too, but I don't think it's a good safe practice (this issue is a living example) to have an input argument that with three different semantics:
- `ceph dashboard sso setup saml2 ... http://example.com/metadata ...` (HTTP URL)
- `ceph dashboard sso setup saml2 ... /home/myuser/metadata.xml ...` (Local file path)
- `ceph dashboard sso setup saml2 ... <?xml=1>...> ...` (XML file contents)

Besides, this command accepts another 2 input files (X509 certificate, and private key), but only as local paths, which is kind of incoherent.

Why not having a single way of providing that data?
`ceph dashboard sso setup saml2 ... /home/myuser/metadata.xml`

BTW, if we unify that, we may modify the type of `idp_metadata` from `CephString` to `CephFilepath`, which already ensures that the input argument is a readable file.

The idea behind printing a `DeprecationWarning` was to allow for a progressive removal of that. But probably it's better to fully remove it right now. As @rjfd is not available for explaining the use cases here, I'd prefer to go with this, and discuss with him whether to remove 2 of these 3 cases.